modules/auth_plugins/lib/open_project/plugins/auth_plugin.rb

```diff
@@ -47,11 +47,23 @@ def self.strategies
     end
 
     def self.providers_for(strategy)
-      strategies[strategy_key(strategy)].map(&:call).flatten.map(&:to_hash)
+      filtered_strategies strategies[strategy_key(strategy)].map(&:call).flatten.map(&:to_hash)
     end
 
     def self.providers
-      strategies.values.flatten.map(&:call).flatten.map(&:to_hash)
+      filtered_strategies strategies.values.flatten.map(&:call).flatten.map(&:to_hash)
+    end
+
+    def self.filtered_strategies(options)
+      options.select do |provider|
+        name = provider[:name]&.to_s
+        next true if !EnterpriseToken.show_banners? || name == 'developer'
+
+        @@warned ||= begin
```

Comment:
This will always return true and include the provider in the result which we do not want.